query_optimizer/ExecutionGenerator.cpp

```diff
@@ -563,12 +572,33 @@ void ExecutionGenerator::convertHashJoin(const P::HashJoinPtr &physical_plan) {
   std::vector<attribute_id> probe_attribute_ids;
   std::vector<attribute_id> build_attribute_ids;
 
+  std::vector<attribute_id> probe_original_attribute_ids;
+  std::vector<attribute_id> build_original_attribute_ids;
+
+  relation_id probe_relation_id;
+  relation_id build_relation_id;
+
   bool any_probe_attributes_nullable = false;
   bool any_build_attributes_nullable = false;
 
+  bool skip_hash_join_optimization = false;
+
   const std::vector<E::AttributeReferencePtr> &left_join_attributes =
       physical_plan->left_join_attributes();
   for (const E::AttributeReferencePtr &left_join_attribute : left_join_attributes) {
+    // Try to determine the original stored relation referenced in the Hash Join.
+    const CatalogRelation *referenced_stored_catalog_relation =
+        optimizer_context_->catalog_database()->getRelationByName(left_join_attribute->relation_name());
+    if (referenced_stored_catalog_relation == nullptr) {
+      // Hash Join optimizations are not possible, if the referenced relation cannot be determined.
+      skip_hash_join_optimization = true;
+    } else {
+      probe_relation_id = referenced_stored_catalog_relation->getID();
+      const attribute_id probe_operator_attribute_id =
+      referenced_stored_catalog_relation->getAttributeByName(left_join_attribute->attribute_name())->getID();
```

Comment:
According to the code style, please add four whitespace indentations.

https://github.com/pivotalsoftware/quickstep/wiki/Quickstep-Style-Guide